### PR TITLE
fixing a bug with duplicte golden resources being returned

### LINKED ISSUE #5084: FindCandidateByExample can return the same resource multiple times
FindCanddiateByExample.findMatchGoldenResourceCandidates can return multiple of the same golden resource. We should prevent this.

### PR DESCRIPTION
**Pull request by TipzCM**
closes #5084

### CODE REVIEW

File: hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/svc/candidate/FindCandidateByExampleSvc.java [hunk 1]
```diff
@@ -78,6 +80,10 @@ protected List<MatchedGoldenResourceCandidate> findMatchGoldenResourceCandidates
 		matchedCandidates = matchedCandidates.stream().filter(mc -> mc.isMatch() || mc.isPossibleMatch()).collect(Collectors.toList());
 		List<String> skippedLogMessages = new ArrayList<>();
 		List<String> matchedLogMessages = new ArrayList<>();
+
+		// we'll track the added ids so we don't add the same resources twice
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/svc/candidate/FindCandidateByExampleSvc.java:84]: How about changing the interface to return a set instead? That could solve potential problems down the line.

**Review by jmarchionatto — COMMENTED** [review]
Question added

**Comment by fil512** [comment]
looks good

File: hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/svc/candidate/FindCandidateByExampleSvc.java (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-mdm/src/main/java/ca/uhn/fhir/jpa/mdm/svc/candidate/FindCandidateByExampleSvc.java:84]: as discussed, will leave as is

**Review by jmarchionatto — APPROVED** [review]

**Pull request merged**